### PR TITLE
RD-5762 Agents widget TS migration

### PR DESCRIPTION
**Pull request by qooban**
## Description

The first PR for RD-5762 brings full TS migration of Agents widget.
In the second PR I'm going to work on ongoing refactors (e.g. translation support and lodash explicit imports).

## Screenshots / Videos
N/A

## Checklist
- [x] My code follows the [UI Code Style](https://cloudifysource.atlassian.net/l/c/x8fq902p).
- [x] I followed the [UI Testing Guidelines](https://cloudifysource.atlassian.net/l/cp/udCVfvrx) and verified that all tests/checks have passed.
- [x] I verified if that change requires any update in the [documentation](https://cloudifysource.atlassian.net/l/c/4XKtPocy).
- [x] I added proper labels to this PR.

## Tests
[Build Status](https://jenkins.cloudify.co/view/UI%20Health%20View/job/Stage-UI-System-Test/4419/) - only `agents_spec`.

## Documentation
N/A

### CODE REVIEW

File: widgets/agents/src/AgentsTable.tsx [hunk 1]
```diff
@@ -1,41 +1,42 @@
...
-    constructor(props: AgentsTableProps, context) {
-        super(props, context);
+export default class AgentsTable extends React.Component<AgentsTableProps, AgentsTableState> {
+    constructor(props: AgentsTableProps) {
+        super(props);
 
         this.state = {
             showModal: false,
-            modal: ''
+            modal: undefined
```
Comment (kubama) [inline widgets/agents/src/AgentsTable.tsx:39]: Does it have any effect to have `undefined` as default value?

File: widgets/agents/src/InstallAgentsModal.tsx [hunk 2]
```diff
@@ -108,9 +111,12 @@ export default function InstallAgentsModal({
         return true;
     }
 
-    function handleInputChange(event, field) {
-        setInputValues({ ...inputValues, ...Stage.Basic.Form.fieldNameValue(field) });
-    }
+    const handleInputChange: NodeFilterProps['onChange'] &
+        StrictDropdownProps['onChange'] &
+        StrictCheckboxProps['onChange'] &
+        StrictInputProps['onChange'] = (_event, field) => {
```
Comment (kubama) [inline widgets/agents/src/InstallAgentsModal.tsx:117]: Maybe `(NodeFilterProps & StrictDropdownProps & StrictCheckboxProps & StrictInputProps)['onChange']` would work?

File: widgets/agents/src/widget.tsx [hunk 3]
```diff
@@ -70,18 +88,18 @@ Stage.defineWidget({
...
         const { Loading } = Stage.Basic;
 
-        if (_.isEmpty(data)) {
+        if (Stage.Utils.isEmptyWidgetData(data)) {
             return <Loading />;
         }
 
-        const params = this.fetchParams(widget, toolbox);
-        const formattedData = {
+        const params = this.fetchParams!(widget, toolbox);
```
Comment (kubama) [inline widgets/agents/src/widget.tsx:98]: So `this.fetchParams` can be nullish? How come?

File: widgets/agents/src/InstallAgentsModal.tsx (hunk 2 above)
Comment (qooban) [reply widgets/agents/src/InstallAgentsModal.tsx]: Yep, it works. Updated.

File: widgets/agents/src/AgentsTable.tsx (hunk 1 above)
Comment (qooban) [reply widgets/agents/src/AgentsTable.tsx]: No, removed.

File: widgets/agents/src/InstallAgentsModal.tsx (hunk 2 above)
Comment (qooban) [reply widgets/agents/src/InstallAgentsModal.tsx]: Right, done.

**Review by kubama — APPROVED** [review]

**Review by Vorbert-Kruk — DISMISSED** [review]
LGTM 💪 
Left one comment, which is not a blocker from my side - that's why I'm approving this PR 🎸

**Review by Vorbert-Kruk — APPROVED** [review]
LGTM 🎉 
Thanks for addressing optional comments!

**Pull request merged**